### PR TITLE
Construct values only when necessary.

### PR DESCRIPTION
**Pull request by AnilBK**
Use properties(size,position etc) directly instead of constructing them again.

And Use the following constructors for Colors, instead of individual values.
https://github.com/godotengine/godot/blob/b8fdeb64678444103a9b6c96ef3ae2c65ad02b2f/core/math/color.h#L236-L244

https://github.com/godotengine/godot/blob/b8fdeb64678444103a9b6c96ef3ae2c65ad02b2f/core/math/color.cpp#L504-L510

Use Image get_size().
https://github.com/godotengine/godot/blob/b1063a7c02c313719fed660feba544dbcd1dac89/core/io/image.cpp#L448-L450

### CODE REVIEW

File: editor/plugins/tiles/tile_map_editor.cpp [hunk 1]
```diff
@@ -790,7 +790,7 @@ void TileMapEditorTilesPlugin::forward_canvas_draw_over_viewport(Control *p_over
 							// Get the tile modulation.
 							Color modulate = tile_data->get_modulate();
 							Color self_modulate = tile_map->get_self_modulate();
-							modulate = Color(modulate.r * self_modulate.r, modulate.g * self_modulate.g, modulate.b * self_modulate.b, modulate.a * self_modulate.a);
+							modulate = modulate * self_modulate;
```
Comment (KoBeWi) [inline editor/plugins/tiles/tile_map_editor.cpp:793]: You could even do `modulate *= self_modulate` here.

Comment (AnilBK) [reply editor/plugins/tiles/tile_map_editor.cpp]: Done.

**Review by KoBeWi — APPROVED** [review]

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Needs a rebase, then should be good to merge.

**Comment by AnilBK** [comment]
Rebased.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**